### PR TITLE
Feature/atr 90 changes in pxcache in event handlers

### PR DESCRIPTION
**Pull request by dropsonic**
Contains two features: ATR-90 and ATR-84.

### CODE REVIEW

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ChangesInPXCache/Walker.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ChangesInPXCache
+{
+	internal class Walker : NestedInvocationWalker
+	{
+		private static readonly ISet<string> MethodNames = new HashSet<string>(StringComparer.Ordinal)
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ChangesInPXCache/Walker.cs]: @dropsonic what about "Clear"/"Remove" methods of PXCache?
Also shouldn't we consider the direct modifications of "Inserted"/"Updated"/"Deleted" collections?

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		private const string PressMethodName = "Press"; // PXAction.Press
+		private const string CachePersistMethodName = "Persist"; // PXCache.Persist
+		private const string CachePersistInserted = "PersistInserted"; // PXCache.PersistInserted
+		private const string CachePersistUpdated = "PersistUpdated"; // PXCache.PersistUpdated
+		private const string CachePersistDeleted = "PersistDeleted"; // PXCache.PersistDeleted
+
+		public static SaveOperationKind GetSaveOperationKind(IMethodSymbol symbol, InvocationExpressionSyntax syntaxNode, 
+			SemanticModel semanticModel, PXContext pxContext)
+		{
+			symbol.ThrowOnNull(nameof (symbol));
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs]: @dropsonic unusual indentation - do you have IDE setting which put space after `nameof`?

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			syntaxNode.ThrowOnNull(nameof (syntaxNode));
+			semanticModel.ThrowOnNull(nameof (semanticModel));
+			pxContext.ThrowOnNull(nameof (pxContext));
+
+			if (symbol.ContainingType != null)
+			{
+				// PXGraph.Actions.PressSave or PXSave.Press
+				if (String.Equals(symbol.Name, PressSaveMethodName, StringComparison.Ordinal)
```
Comment (SENya1990) [inline src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs]: @dropsonic are String.Equals necessary here? We are not working with UI strings and could rely on standard `==` which is ordinal by default and is shorter

Comment (SENya1990) [reply src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs]: @dropsonic also isn't a swith with pattern matching a little more readable?

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ChangesInPXCache/Walker.cs (hunk 1 above)
Comment (dropsonic) [reply src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ChangesInPXCache/Walker.cs]: See the item in JIRA - I've asked Ruslan about that.
`Inserted` / `Updated` / `Deleted` are not collections, they are properties which return `IEnumerable` from the inner private collection which is inaccessible for direct modifications.

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs (hunk 2 above)
Comment (dropsonic) [reply src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs]: It is a default setting from Resharper (for `nameof` and `typeof`).

File: src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs (hunk 3 above)
Comment (dropsonic) [reply src/Acuminator/Acuminator.Analyzers/StaticAnalysis/SavingChanges/SaveOperationHelper.cs]: @SENya1990 you're completely right, I'll fix it

**Pull request merged**